### PR TITLE
Add emoji.gg presence

### PR DESCRIPTION
**Pull request by dtmcl**
Adds an emoji.gg presence

image
image
image
image
image

### CODE REVIEW

File: websites/E/Emoji.gg/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/dist/metadata.json:2]: Outdated schema
```suggestion
  "$schema": "https://schemas.premid.app/metadata/1.5",
```

File: websites/E/Emoji.gg/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "service": "Emoji.gg",
+  "author": {
+    "name": "Kohai",
+    "id": "116218776495587329"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/dDhx5sD.png",
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/dist/metadata.json:9]: Not 512x

File: websites/E/Emoji.gg/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "thumbnail": "https://i.imgur.com/Xgx0O4O.png",
+  "version": "1.0.0",
+  "color": "#971ce9",
+  "tags": [
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/dist/metadata.json:13]: add a `nsfw` tag for the nsfw category

File: websites/E/Emoji.gg/presence.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  title: HTMLElement,
+  replace: HTMLElement,
+  search: HTMLElement;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "emojigg"
+  };
+
+  if (document.location.hostname === "emoji.gg") {
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/presence.ts:15]: Remove this if statement

File: websites/E/Emoji.gg/presence.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    if (document.location.pathname === "/") {
+      presenceData.startTimestamp = browsingStamp;
+      presenceData.details = "Viewing home page";
+      presenceData.state = "Latest Emojis";
+    } else if (document.location.pathname.includes("/emoji/")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/presence.ts:21]: set startTimestamp in the PresenceData object

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
fix indentation too

File: websites/E/Emoji.gg/presence.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
+const presence = new Presence({
+    clientId: "387040638249467904",
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+let user: HTMLElement,
+  title: HTMLElement,
+  replace: HTMLElement,
+  search: HTMLElement;
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/presence.ts:9]: Why do you have these variables when they're never used?

File: websites/E/Emoji.gg/presence.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        label: "View Profile",
+        url: document.URL,
+      },
+    ];
+  } else if (document.location.pathname.includes("/pack/")) {
+    title = document.querySelector("h1");
+    user = document.querySelector(".user-stats > a");
+
+    presenceData.details = "Viewing " + title.innerText;
+    presenceData.state = "by " + user.innerText;
```
Comment (EncryptedDev) [inline websites/E/Emoji.gg/presence.ts:59]: Use proper string contactination

**Comment by Slowlife01** [comment]
Inactive

**Pull request closed without merging**